divide_and_conquer/binary_exponentiation.cpp

```diff
@@ -0,0 +1,90 @@
+/**
+ * @file
+ * @brief Implementation of [Binary Exponentiation]
+ * (https://cp-algorithms.com/algebra/binary-exp.html)
+ * @details
+ * Given a base and an exponent, we want to return the resulting value.
+ * Normal exponentiation calculates a^n in O(n), but using binary
+ * exponentiation we are able to calculate a^n in O(logn).
+ * @author [Mason Green](https://github.com/masongreen30)
+ */
+
+#include <cassert>
+#include <iostream>
```

Comment:
Please add a one-line description of what the library/header is for (**see the example below**).

```c++
#include <iostream>    /// for IO operations
#include <cassert>    /// for assert
```